distributed/preloading.py

```diff
@@ -111,65 +131,59 @@ def _import_module(name, file_dir=None):
     }
 
 
-def on_creation(names, file_dir: str = None) -> dict:
-    """ Imports each of the preload modules
-
-    Parameters
-    ----------
-    names: list of strings
-        Module names or file paths
-    file_dir: string
-        Path of a directory where files should be copied
+class Preload:
     """
-    if isinstance(names, str):
-        names = [names]
-
-    return {name: _import_module(name, file_dir=file_dir) for name in names}
-
-
-async def on_start(modules: dict, dask_server=None, argv=None):
-    """ Run when the server finishes its start method
 
     Parameters
     ----------
-    modules: Dict[str, module]
-        The imported modules, from on_creation
     dask_server: dask.distributed.Server
         The Worker or Scheduler
     argv: [string]
         List of string arguments passed to click-configurable `dask_setup`.
     file_dir: string
         Path of a directory where files should be copied
     """
-    for name, interface in modules.items():
-        dask_setup = interface.get("dask_setup", None)
+
+    def __init__(self, dask_server, name, argv, file_dir):
+        self.dask_server = dask_server
+        self.name = name
+        self.argv = argv
+        self.file_dir = file_dir
+
+        self.module = _import_module(name, file_dir)
```

Comment:
I would move this to under `start`, no reason to do it here afaict.

Reply:
In between now and start lots of decisions are made.  For example, it's useful to import the module before we decide on protocol (this is an old request from the ucx folks, who want to `import ucx` to depend on the side effects of loading `"ucx://"` into the registry of protocols before we try to use that protocol.

Reply:
Then maybe set this to `None` in the case of a remote address? Creating and stashing coroutine that is later run seems like a bit convoluted logic that may be cleaner if the logic wasn't split between `__init__` and `start`. idk.

Reply:
Yeah, I agree that it's a little convoluted.  Worse is that behavior for web-downloaded preloads is different than those for file-loaded ones (we don't get on-import behavior for web-downloaded preloads).  It's a choice between this and poor testing though, and at least for now I'd prefer to optimize for testability.

In terms of how to handle the magic coroutine I don't really have a good answer here, other than to store state about if it's a web address, and to explicitly call this out in `def start`.  I'll do that now.  I guess we're being more explicit, which is good.

Reply:
OK, we're now more explicit.  The behavior is the same, but things are hopefully a bit cleaner.